lib/mailchimp_api/resources/interest_category.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module MailchimpAPI
+  class InterestCategory < Base
```

Comment:
Can this have a `belongs_to :list`?

Reply:
I can't get this to work. One or multiple of [I suck, ActiveResource docs suck, It's not meant to work that way].

When the InterestCategory is parsed, ActiveResource actually _removes_ `list_id` from its attributes and leaves it in `prefix_options`. This means when I call `interest_category.list`, it tries to get foreign key by calling `list_id`, which was forcefully removed.

I've tried changing the path substitution parameter from `:list_id` to just `:list` just so that it doesn't touch the `list_id` attribute, but this causes a separate issue when calling `list.interest_categories` because it does this when loading the InterestCategories:

```
reflection.klass.find(:all, params: { :"#{self.class.element_name}_id" => self.id })
```

which resolves to `:list_id`. It seemed silly and confusing to change the `element_name` of List to something else...